src/rgw/rgw_aws_lambda.cc

```diff
@@ -0,0 +1,209 @@
+//
+// Created by prateek on 30/03/20.
+//
+
+#include <string>
+#include <iostream>
+#include <utility>
+#include "common/dout.h"
+#include <aws/core/Aws.h>
+#include <aws/core/utils/Outcome.h>
+#include <aws/lambda/LambdaClient.h>
+#include <aws/lambda/model/InvokeRequest.h>
+#include <boost/lockfree/queue.hpp>
+#include <aws/core/utils/ARN.h>
+#include <aws/core/auth/AWSCredentials.h>
+#include <unordered_set>
+
+
+namespace rgw::lambda {
+  static const int STATUS_OK = 1;
+  static const int STATUS_FULL = 2;
+  static const int STATUS_MANAGER_STOPPED = 3;
+  static const int STATUS_CLIENT_CLOSED = 4;
+
+  struct client_id_t {
+    const std::string &accessKey;
+    const std::string &accessSecret;
+    const std::string &arn;
+    bool useSSL;
+
+    client_id_t(const std::string &accessKey, const std::string &accessSecret,
+                const std::string &arn, bool useSsl) : accessKey(accessKey),
+                                                       accessSecret(accessSecret),
+                                                       arn(arn), useSSL(useSsl) {}
+
+    bool operator==(const client_id_t &other) const {
+      return accessKey == other.accessKey &&
+             accessSecret == other.accessSecret &&
+             arn == other.arn && useSSL == other.useSSL;
+    }
+
+    struct hasher {
+      std::size_t operator()(const client_id_t &k) const {
+        return ((std::hash<std::string>()(k.accessKey)
+                 ^ (std::hash<std::string>()(k.accessSecret) << 1)) >> 1)
+               ^ (std::hash<std::string>()(k.arn) << 1)
+               ^ (std::hash<bool>()(k.useSSL) << 1);
+      }
+    };
+
+
+  };
+
+  struct message_t {
+    const Aws::Lambda::LambdaClient *client;
+    const std::string function_name;
+    const std::string payload;
+
+    message_t(const Aws::Lambda::LambdaClient *client, std::string function_name, std::string payload) :
+            client(client),
+            function_name(std::move(function_name)),
+            payload(std::move(payload)) {};
+  };
+
+  typedef std::unordered_map<client_id_t, Aws::Lambda::LambdaClient *, client_id_t::hasher> ClientList;
+
+
+  typedef boost::lockfree::queue<message_t *, boost::lockfree::fixed_sized<true>> MessageQueue;
+
+  class Manager {
+  private:
+    MessageQueue messageQueue;
+    ClientList clientList;
+    std::thread runner;
+    bool stopped;
+    mutable std::mutex clientsLock;
+    CephContext *const cct;
+
+    void publish_internal(message_t *msg) {
+      const std::unique_ptr<message_t> msg_owner(msg);
+      std::cout << "Invoking.." << std::endl;
+      const Aws::Lambda::LambdaClient *client = msg->client;
+      Aws::Lambda::Model::InvokeRequest invokeRequest;
+      invokeRequest.SetInvocationType(Aws::Lambda::Model::InvocationType::RequestResponse);
```

Comment:
i dont think we do anything with the response
can we set the type to be only "request"?